app/main.py

```diff
@@ -1,13 +1,25 @@
 def format_linter_error(error: dict) -> dict:
-    # white your code here
-    pass
+    return {"line": error["line_number"],
+            "column": error["column_number"], "message": error["text"],
+            "name": error["code"], "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # white your code here
-    pass
+    return {"errors": [{"line": error["line_number"],
+                        "column": error["column_number"],
+                        "message": error["text"],
+                        "name": error["code"], "source": "flake8"}
+                       for error in errors],
+            "path": file_path,
+            "status": "passed" if len(errors) == 0 else "failed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # white your code here
-    pass
+    return [{"errors": [{"line": error["line_number"],
+                         "column": error["column_number"],
+                         "message": error["text"],
+                         "name": error["code"], "source": "flake8"}
+                        for error in errors],
+             "path": file_path,
+             "status": "passed" if len(errors) == 0 else "failed"}
```

Comment:
The same as previous comment, don't repeat yourself! Almost all work you do in 'format_single_linter_file'.